app/models/lead.rb

```diff
@@ -0,0 +1,3 @@
+class Lead <ApplicationRecord
+
```

Comment:
does there need to be any validation here? On schools, I am validating the SF id
https://github.com/openstax/openstax-salesforce-api/blob/master/app/models/school.rb#L2

Reply:
@mwharrison There is no id associated with Leads. I could validate first and last name. Since we are pulling the data from Salesforce, why the validation?

Reply:
@edwoodward The suggestion for validation came from JP on my first PR - https://github.com/openstax/openstax-salesforce-api/pull/2#discussion_r444496199
Best practice from what I read is to add to the model and the database.

There is a id on a Salesforce lead and it would probably be the best unique identifier to make sure we are modifying the correct lead when the time comes. Here is a screenshot of a quick query I did on the Salesforce developer portal. 

![Screen Shot 2020-07-15 at 1 50 33 PM](https://user-images.githubusercontent.com/3905516/87583835-28d0de00-c6a2-11ea-9f8c-e643a3e7a8d9.png)

Reply:
@mwharrison That id does not come through openstax_salesforce. See https://github.com/openstax/openstax_salesforce/blob/master/lib/openstax/salesforce/remote/lead.rb

Reply:
@mwharrison There is no id field on Schools in openstax_salesforce. How is it getting populated?

Reply:
@edwoodward - I didn't look at openstax_salesforce gem (since that wasn't installed yet) but I think we'll need to add these fields unless there is some other reliable way we can make sure we are consistently dealing with the same records.

Reply:
@mwharrison Accounts matches Leads via email address which is what I did in my code. None of the objects in openstax_salesforce have an id. Let me look at the Accounts code and see how a user gets a salesforce id. You are right, we probably should add the salesforce field to all of the objects in openstax_salesforce.

Reply:
@mwharrison An Accounts user gets their id from the Contact. Accounts does a query from salesforce using openstax_salesforce which includes the id. See https://github.com/openstax/accounts/blob/751f5859566fb6df9061bad33049573c73b4aecf/app/routines/update_user_salesforce_info.rb#L164

Reply:
@mwharrison  I added the salesforce_id to Leads and added the validation.